packages/editor/src/components/post-url/index.js

```diff
@@ -6,122 +6,146 @@ import { safeDecodeURIComponent, cleanForSlug } from '@wordpress/url';
-			),
-			viewPostLabel: postType?.labels.view_item,
-			postLink: post.link,
-			permalinkPrefix: permalinkParts?.prefix,
-			permalinkSuffix: permalinkParts?.suffix,
-		};
-	}, [] );
+	const { isEditable, postSlug, postLink, permalinkPrefix, permalinkSuffix } =
+		useSelect( ( select ) => {
+			const post = select( editorStore ).getCurrentPost();
+			const postTypeSlug = select( editorStore ).getCurrentPostType();
+			const postType = select( coreStore ).getPostType( postTypeSlug );
+			const permalinkParts = select( editorStore ).getPermalinkParts();
+			const hasPublishAction =
+				post?._links?.[ 'wp:action-publish' ] ?? false;
 
+			return {
+				isEditable:
+					select( editorStore ).isPermalinkEditable() &&
+					hasPublishAction,
+				postSlug: safeDecodeURIComponent(
+					select( editorStore ).getEditedPostSlug()
+				),
+				viewPostLabel: postType?.labels.view_item,
+				postLink: post.link,
+				permalinkPrefix: permalinkParts?.prefix,
+				permalinkSuffix: permalinkParts?.suffix,
+			};
+		}, [] );
 	const { editPost } = useDispatch( editorStore );
-
+	const { createNotice } = useDispatch( noticesStore );
 	const [ forceEmptyField, setForceEmptyField ] = useState( false );
-
+	const postUrlLabel = usePostURLLabel();
+	const copyButtonRef = useCopyToClipboard( postUrlLabel, () => {
+		createNotice( 'info', __( 'Copied URL to clipboard.' ), {
+			isDismissible: true,
+			type: 'snackbar',
+		} );
+	} );
 	return (
 		<div className="editor-post-url">
-			<InspectorPopoverHeader title={ __( 'URL' ) } onClose={ onClose } />
-			{ isEditable && (
-				<TextControl
-					__nextHasNoMarginBottom
-					label={ __( 'Permalink' ) }
-					value={ forceEmptyField ? '' : postSlug }
-					autoComplete="off"
-					spellCheck="false"
-					help={
-						<>
-							{ __( 'The last part of the URL.' ) }{ ' ' }
-							<ExternalLink
-								href={ __(
-									'https://wordpress.org/documentation/article/page-post-settings-sidebar/#permalink'
-								) }
-							>
-								{ __( 'Learn more.' ) }
-							</ExternalLink>
-						</>
-					}
-					onChange={ ( newValue ) => {
-						editPost( { slug: newValue } );
-						// When we delete the field the permalink gets
-						// reverted to the original value.
-						// The forceEmptyField logic allows the user to have
-						// the field temporarily empty while typing.
-						if ( ! newValue ) {
-							if ( ! forceEmptyField ) {
-								setForceEmptyField( true );
+			<InspectorPopoverHeader
+				title={ __( 'Slug' ) }
+				onClose={ onClose }
+			/>
+			<VStack spacing={ 3 }>
+				<div>
+					{ __(
+						'The slug forms the last part of the URL for a page. '
+					) }
+					<ExternalLink
+						href={ __(
+							'https://wordpress.org/documentation/article/page-post-settings-sidebar/#permalink'
+						) }
+					>
+						{ __( 'Learn more.' ) }
+					</ExternalLink>
+				</div>
+				<div>
+					{ isEditable && (
+						<InputControl
+							__next40pxDefaultSize
```

Comment:
This is probably the most noticeable difference. I replaced `TextControl` with `InputControl`. This way we also have the prefix/suffix support.